### PR TITLE
stmmac: Disable EEE to avoid rare network outage

### PR DESCRIPTION
**Pull request by Ray-future**
Fixes an issue on WP2 where Network disconnects unexpectedly on Gigabit networks.

This is not a fix but a workaround which disables EEE completely.
It's probably a good idea to fix revert that commit someday once the EEE
timer issue is found and fixed. Since I can't reproduce that issue myself it's best to disable EEE.

Thanks **@jester-xbmc** for finding that issue and doing all the research for a "fix".

Not sure if it's a proper fix. Maybe it's better to do a project specific patch in LE?
It should be tested before merge.
So far I did iperf3 tests. Speed is not effected as far I can tell and the connection is stable.

Reference:
https://forum.libreelec.tv/thread-3214-post-28570.html#pid28570
http://lists.infradead.org/pipermail/linux-amlogic/2016-November/001538.html

### CODE REVIEW

**Comment by jester-xbmc** [comment]
to add my 2cents:

This has been an "issue" on both WP2 (probably Hub too) and C2 (both S905) to make this patch project specific could be an option but I think its fine as is.
EEE isnt a big "loss" imho, also it seems that S905 has been added to mainline kernel (4.10) so hopefully we can switch mainline or aml provides a new kernel which updates this driver (as its very old, I tried cherrypicking a newer one but ran into issues compiling -> https://github.com/jester-xbmc/linux-amlogic/commit/9a57775ea55f674df941a3bffec58e99223fe081

In short, the patch is confirmed working and from my point of view is ready for LE (tested with raybuntu built, im compiling from master with this patch right now to confirm on 64bit too.)

**Comment by Ray-future** [comment]
Odroid C2 is using another network driver :). Its a RTL8211F.

**Comment by jester-xbmc** [comment]
True, same eee issue strangely.
I'm running 64bit build from master right now with the patch, no issues so far, will report back later.

**Comment by jester-xbmc** [comment]
All ok on 64 bit,this can be merged.

now to find that pesky memleak ;)

**Comment by Ray-future** [comment]
Not sure it this should be merged that fast. It should be reviewed if there are side effects or if its better to fix the EEE timers.

**Comment by jester-xbmc** [comment]
Its running with several users without issue at the moment, so apart from disabling eee, there seems to be no down side, id agree with fixing it but there are better drivers once we upgrade the kernel, so a wasted effort imho.

**Comment by Ray-future** [comment]
Right, but I guess we'll have to wait until this is reviewed, approved and merged. I'm still not sure if it's not better to have a project specific patch in LE.

**Review by jester-xbmc — APPROVED** [review]

**Review by codesnake — APPROVED** [review]
Looks good to me. Thanks @Raybuntu.

**Comment by Ray-future** [comment]
Thanks @codesnake for the review and @kwiboo for the merge.
